lib/internal/util.js

```diff
@@ -176,9 +176,15 @@ function slowCases(enc) {
 }
 
 function emitExperimentalWarning(feature) {
+  const { getOptionValue } = require('internal/options');
+  const warningoption = getOptionValue('--no-experimental-warnings');
   if (experimentalWarnings.has(feature)) return;
+  if (warningoption === feature) {
+    experimentalWarnings.add(feature);
+    return;
+  }
   const msg = `${feature} is an experimental feature. This feature could ` +
-       'change at any time';
+  'change at any time';
```

Comment:
I think left-aligning this with the beginning of the string on the previous line like this:

```suggestion
              'change at any time';
```

will read better and will look less like a mistake or some kind of `'use strict';`-like directive.